lib/resources/mssql_session.rb

```diff
@@ -29,15 +29,18 @@ class MssqlSession < Inspec.resource(1)
       end
     "
 
-    attr_reader :user, :password, :host, :port, :instance
+    attr_reader :user, :password, :host, :port, :instance, :local_mode
     def initialize(opts = {})
       @user = opts[:user]
       @password = opts[:password] || opts[:pass]
       if opts[:pass]
         warn '[DEPRECATED] use `password` option to supply password instead of `pass`'
       end
-      @host = opts[:host] || 'localhost'
-      @port = opts[:port] || '1433'
+      @local_mode = opts[:local_mode]
+      unless local_mode?
+        @host = opts[:host] || 'localhost'
+        @port = opts[:port]
```

Comment:
While this is probably fine as `sqlcmd` defaults to `1433` I think i'd like to keep the explicit set incase we change what this uses under the hood.

Reply:
We'd want to retain the ability to set port to nil to support dynamic ports though. Forgive my ignorance, but does ruby have an equivalent of `$PSBoundParameters` from powershell so we can tell if a parameter is explicitly passed?

Reply:
Something like:
```
opts[:port] = 1433 unless opts.key?(:port)
```
should only set it if the key is not previously defined.